APTITUDE/APTITUDETEST.Data/Data/TestRepository.cs

```diff
@@ -165,6 +165,126 @@ public async Task<JsonResult> UpdateTestGroup(UpdateTestGroupVM updateTest)
+                        Result = true,
+                        StatusCode = ResponseStatusCode.OK
+                    });
+                }
+
+                #region Code to check questions are available in DB
+                Func<TestQuestionsCountVM, int> func = x => x.OneMarkQuestion;
+                var MarkQuestionCountReq = 0;
+                for (int questionType = 1; questionType <= 2; questionType++)
+                {
+                    for (int i = 1; i <= 5; i++)
+                    {
+                        int MarkQuestionCountInDB = await Task.FromResult(_context.Questions.Where(t => t.Topic == addTestQuestion.TopicId && t.QuestionType == questionType && t.Difficulty == i && t.IsDeleted == false).Count());
+                        switch (i)
+                        {
+                            case 1:
+                                func = x => x.OneMarkQuestion;
+                                break;
+                            case 2:
+                                func = x => x.TwoMarkQuestion;
+                                break;
+                            case 3:
+                                func = x => x.ThreeMarkQuestion;
+                                break;
+                            case 4:
+                                func = x => x.FourMarkQuestion;
+                                break;
+                            case 5:
+                                func = x => x.FiveMarkQuestion;
+                                break;
+                            default:
+                                break;
+                        }
+
+                        MarkQuestionCountReq = addTestQuestion.TestQuestionsCount.Where(t => t.QuestionType == questionType).Select(func).FirstOrDefault();
+
+                        if (MarkQuestionCountReq > MarkQuestionCountInDB)
+                        {
+                            return new JsonResult(new ApiResponse<string>
+                            {
+                                Message = string.Format(ResponseMessages.NotEnoughQuestion, i),
+                                Result = true,
+                                StatusCode = ResponseStatusCode.OK
+                            });
+                        }
+                    }
+                }
+                #endregion
+                
+                TestQuestions testQuestion = await Task.FromResult(_context.TestQuestions.Where(t => t.TestId == addTestQuestion.TestId && t.TopicId == addTestQuestion.TopicId && t.IsDeleted == false).FirstOrDefault());
+                if (testQuestion != null)
+                {
+                    return new JsonResult(new ApiResponse<string>
+                    {
+                        Message = string.Format(ResponseMessages.TestTopicAlreadyExists),
+                        Result = true,
+                        StatusCode = ResponseStatusCode.OK
+                    });
+                }
+
```

Comment:
include validation to match addTestQuestion.NoOfQuestion and total questions count added based on question type